shared/dev/log-send/render.native.js

```diff
@@ -1,26 +1,43 @@
 // @flow
+import RNFS from 'react-native-fs'
 import React, {Component} from 'react'
-import type {Props} from './render'
 import {Box, Text, Button, NativeLinking, NativeClipboard} from '../../common-adapters/index.native'
+import {dumpLoggers} from '../../util/periodic-logger'
 import {globalStyles} from '../../styles'
 
-type State = {
-  copiedToClipboard: boolean,
-}
-
-class LogSendRender extends Component<void, Props, State> {
-  state: State;
+import type {Props} from './render'
 
-  constructor (props: Props) {
-    super(props)
-    this.state = {copiedToClipboard: false}
+class LogSendRender extends Component<void, Props, {copiedToClipboard: boolean}> {
+  state = {
+    copiedToClipboard: false,
   }
 
-  _copyToClipboard () {
+  _copyToClipboard = () => {
     NativeClipboard.setString(this.props.logSendId || '')
     this.setState({copiedToClipboard: true})
   }
 
+  _logSend = () => {
+    // We don't get the notification from the daemon so we have to do this ourselves
+    const logs = []
+    dumpLoggers((...args) => {
```

Comment:
When the user clicks log send we dump the period logger state and write it to a file on the device